File: src/main/java/beans/ServerToChannel.java
package beans;

/**
 * Classe di appoggio per ottenere serverID e channelID
 * per i publish del feed RSS
 */
public class ServerToChannel {

    private String serverID;
    private String channelID;

    public ServerToChannel(String serverID, String channelID) {
        setServerID(serverID);
        setChannelID(channelID);
    }

    /**
     * Metodo per ottenere il serverID
     *
     * @return serverID
     */
    public String getServerID() {
        return serverID;
    }

    /**
     * Metodo per impostare il serverID
     *
     * @param serverID id del server discord
     */

    public void setServerID(String serverID) {
        this.serverID = serverID;
    }

    /**
     * Metodo per ottenere il channelID
     *
     * @return String id del canale
     */
    public String getChannelID() {
        return channelID;
    }

    /**
     * Metodo per impostare il channelID
     *
     * @param channelID id del canale testuale
     */
    public void setChannelID(String channelID) {
        this.channelID = channelID;
    }


    @Override
    public boolean equals(Object object) {
        if (object instanceof ServerToChannel) {

            ServerToChannel obj = (ServerToChannel) object;

            if (obj.serverID != null) {
                if (obj.serverID.equals(this.serverID)) {
                    return true;
                } else {
                    return false;
                }
            } else {
                return false;
            }
        } else {
            return false;
        }
    }
}

File: src/main/kotlin/it/discordbot/beans/ServerToChannel.kt
package it.discordbot.beans

/**
 * Classe contenitore degli id dei server e relativi canali per
 * le pubblicazioni schedulate
 * @property serverID String id del server
 * @property channelID String id del canale
 * @constructor
 */
data class ServerToChannel(val serverID: String, val channelID: String) {
	override fun equals(other: Any?): Boolean {
		return if (other != null) {
			if (other is ServerToChannel) {
				other.serverID == this.serverID
			} else {
				false
			}
		} else {
			false
		}

	}

	override fun hashCode(): Int {
		var result = serverID.hashCode()
		result = 31 * result + channelID.hashCode()
		return result
	}
}